File: src/main/java/com/orellana/products/DTO/TransaccionesDTO.java
package com.orellana.products.DTO;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.orellana.products.Enums.StatusTransaccion;
import com.orellana.products.Enums.TipoTransaccion;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class TransaccionesDTO {

    private Long idTransaccion;

    private Integer totalProductos;

    private BigDecimal totalPrecio;

    private TipoTransaccion tipoTransaccion;

    private StatusTransaccion status;

    private String description;

    private LocalDateTime updatedAt;

    private LocalDateTime createdAt;

    private UserDTO user;

    private ProductsDTO product;

    private LocalesDTO local;

}

File: frontend/src/data/DTOs/TransaccionesDTO.ts
export type TransaccionesDTO = {
    id: number;
    totalProductos: number;
    totalPrecio: string; 
    tipoTransaccion: string;
    status: string;
    description?: string; 
    updatedAt: Date; 
    createdAt: Date;
    user: number;
    product: number;
    local: number;
};

export type TransaccionRequestDTO = {
    idProduct: number;
    quantity: number;
    description: string;
    idLocal: number;
};